pkg/checks/traceroute/traceroute.go

```diff
@@ -58,15 +66,46 @@ func tcpHop(ctx context.Context, addr net.Addr, ttl int, timeout time.Duration)
 			},
 		}
 
+		span.AddEvent("Attempting TCP connection", trace.WithAttributes(
+			attribute.String("remote_addr", addr.String()),
+			attribute.Int("ttl", ttl),
+			attribute.Int("port", port),
+		))
+
 		// Attempt to connect to the target host
 		conn, err := dialer.DialContext(ctx, "tcp", addr.String())
-		if !errors.Is(err, unix.EADDRINUSE) {
+
+		switch {
+		case err == nil:
+			span.AddEvent("TCP connection succeeded", trace.WithAttributes(
+				attribute.Stringer("remote_addr", addr),
+				attribute.Int("ttl", ttl),
+				attribute.Int("port", port),
+			))
+			return conn, port, nil
+		case errors.Is(err, unix.EADDRINUSE):
+			// Address in use, retry by continuing the loop
+			continue
+		case errors.Is(err, syscall.EHOSTUNREACH):
+			// No route to host is no error because of how tcp traceroute works
+			// we are expecting the connection to fail because of TTL expiry
+			span.SetStatus(codes.Unset, "No route to host")
+			span.AddEvent("No route to host", trace.WithAttributes(
+				attribute.String("error", err.Error()),
+			))
+			logger.FromContext(ctx).DebugContext(ctx, "No route to host", "error", err.Error())
 			return conn, port, err
```

Comment:
Is this an error or not? I'm confused now :D If it's not an error, then please return nil as the third argument. If it's an error, then we mark should mark the span as a faulty one.

Reply:
No need for a custom error, as special errors are typical in file and network operations.

Since the hop hasn't succeeded and there is indeed an error, shouldn't we mark the span as an error span as well? We need to mark each time an error happens, so it's easier to read the spans in the backend UI tool displaying them (Jaeger for example).

Reply:
I've changed this to be displayed as error instead of `codes.Unset`